odoo/addons/base/models/ir_mail_server.py

```diff
@@ -273,17 +303,25 @@ def test_smtp_connection(self):
                     # ignored, just a consequence of the previous exception
                     pass
 
-        message = _("Connection Test Successful!")
+        messages = [_('Connection Test Successful!')]
+        if autodetect_max_email_size and self:
```

Comment:
Maybe we can be defensive
```suggestion
        if autodetect_max_email_size and self.max_email_size:
```